pkg/api/admin/platformworkloadidentityroleset.go

```diff
@@ -43,4 +43,13 @@ type PlatformWorkloadIdentityRole struct {
 
 	// ServiceAccounts represents the set of service accounts associated with the given operator, since each service account needs its own federated credential.
 	ServiceAccounts []string `json:"serviceAccounts,omitempty" mutable:"true" validate:"required"`
+
+	// SecretLocation represents the location of the in-cluster secret containing credentials for the platform workload identity.
+	SecretLocation SecretLocation `json:"secretLocation,omitempty"`
+}
+
+// SecretLocation represents the location of the in-cluster secret containing credentials for the platform workload identity.
+type SecretLocation struct {
+	Namespace string `json:"namespace,omitempty"`
+	Name      string `json:"name,omitempty"`
```

Comment:
I'll add them, I don't think it would harm anything to include them as we expect these fields to always be present for every identity, and we should allow ourselves the ability to change them in the future just in case.